### ISSUE TITLE
False positive ConstantConditionalOperator reported on snprintf usage

### ISSUE THREAD

**Issue by wfleming** [oclint/oclint#313]
I'm seeing "constant conditional operator" reported on many usages of `stdio.h` functions in my code, in places that I'm not using the ternary operator. I've found it be reproducible on this small test case (`oclint-test.c`):

``` c
#include <stdio.h>
#include <stdlib.h>

void get_data_dir(char *buf, int buflen);

void get_data_dir(char *buf, int buflen)
{
  snprintf(buf, buflen, "%s/.appname", getenv("HOME"));
}
```

When I run `oclint oclint-test.c --`, I see:

```
OCLint Report

Summary: TotalFiles=1 FilesWithViolations=1 P1=0 P2=1 P3=0

/Users/will/projects/oclint-test/oclint-test.c:8:3: constant conditional operator [basic|P2]

[OCLint (http://oclint.org) v0.10.2]
```

`oclint --version` outputs:

```
LLVM (http://llvm.org/):
  LLVM version 3.7.0
  Optimized build.
  Built Dec 28 2015 (09:59:04).
  Default target: x86_64-apple-darwin15.0.0
  Host CPU: broadwell

OCLint (http://oclint.org/):
  OCLint version 0.10.2.
  Built Dec 28 2015 (10:15:51).
```

I'm not sure if I've misunderstood something elementary, but this was surprising to me.

**Comment by UnrealQuester** [comment]
I am unable to reproduce this behaviour on my linux machine. Looking at your oclint version I am assuming this might be a osx specific issue.

**Comment by wfleming** [comment]
Seems likely to be an OS X issue, then, yeah: I was also unable to reproduce on an Ubuntu VM. I was using the 0.10.2 binary installed via Homebrew on OS X originally, but also tried building from source on master and saw the same behavior.

Is there a way for me to provide more detailed debugging information to help track this down? Or, if I have time to dig into this deeper myself, do you have any suggestions of where in the code I might start digging?

**Comment by UnrealQuester** [comment]
The violation itself is in [ConstantConditionalOperatorRule.cpp](https://github.com/oclint/oclint/blob/master/oclint-rules/rules/basic/ConstantConditionalOperatorRule.cpp#L32). You may want to add some debug output there to see if the method gets called. Maybe snprintf is a macro (or gets inlined and then traversed by clang) and contains a constant conditional expression.

**Comment by bok** [comment]
snprintf might be a macro if you compile with optimizations activated. Try with `-O0`.

**Comment by AbigailBuccaneer** [comment]
Assuming the cause is that `snprintf` is a macro, #378 would fix this.

**Comment by ryuichis** [comment]
> #378 would fix this.
